File: code/python/singleton/Library.py
class Library:
    __instance = None

    def __init__(self):
        pass

    @staticmethod
    def get_instance():
        if Library.__instance is None:
            Library.__instance = Library()
        return Library.__instance

    def borrow_a_book(self):
        print("Borrowing a book from the library...")

File: code/typescript/src/singleton/Library.ts
export class Library {
    private static uniqueInstance: Library;

    private constructor() { }

    static getInstance(): Library {
        if (!Library.uniqueInstance) {
            Library.uniqueInstance = new Library();
            console.log("Granting access to the central library");
        }
        return Library.uniqueInstance;
    }

    borrowABook(): void {
        console.log("Borrowing a book from the library...");
    }
}
